pkg/monitoring/rules/rules-tests.yaml

```diff
@@ -0,0 +1,192 @@
+# Unit tests for the prometheus rules
+rule_files:
+  - rules.json
+
+tests:
+  # SSPDown alert tests
+  - interval: "1m"
+    input_series:
+      - series: 'up{pod="ssp-operator-12345"}'
+        values: '0x5 1'
+
+    alert_rule_test:
+      - eval_time: "5m"
+        alertname: "SSPDown"
+        exp_alerts:
+          - exp_annotations:
+              summary: "All SSP operator pods are down."
+              runbook_url: "test-runbook:SSPDown"
+            exp_labels:
+              severity: "critical"
+              operator_health_impact: "critical"
+              kubernetes_operator_part_of: "kubevirt"
+              kubernetes_operator_component: "ssp-operator"
+
+      - eval_time: "6m"
+        alertname: "SSPDown"
+        exp_alerts: []
+
+  # SSPTemplateValidatorDown alert tests
+  - interval: "1m"
+    input_series:
+      - series: 'up{pod="virt-template-validator-12345"}'
+        values: '0x5 1'
+
+    alert_rule_test:
+      - eval_time: "5m"
+        alertname: "SSPTemplateValidatorDown"
+        exp_alerts:
+          - exp_annotations:
+              summary: "All Template Validator pods are down."
+              runbook_url: "test-runbook:SSPTemplateValidatorDown"
+            exp_labels:
+              severity: "critical"
+              operator_health_impact: "critical"
+              kubernetes_operator_part_of: "kubevirt"
+              kubernetes_operator_component: "ssp-operator"
+
+      - eval_time: "6m"
+        alertname: "SSPTemplateValidatorDown"
+        exp_alerts: [ ]
+
+  # SSPFailingToReconcile alert tests
+  - interval: "1m"
+    input_series:
+      - series: 'up{pod="ssp-operator-12345"}'
+        values: '0x5 1x5 1'
+      - series: 'kubevirt_ssp_operator_reconcile_succeeded{pod="ssp-operator-12345"}'
+        values: '0x5 0x5 1'
+
+    alert_rule_test:
+      # SSP pod is down -> should not trigger SSPFailingToReconcile alert
+      - eval_time: "5m"
+        alertname: "SSPFailingToReconcile"
+        exp_alerts: []
+
+      # SSP pod is up, but failed to reconcile
+      - eval_time: "11m"
+        alertname: "SSPFailingToReconcile"
+        exp_alerts:
+          - exp_annotations:
+              summary: "The ssp-operator pod is up but failing to reconcile"
+              runbook_url: "test-runbook:SSPFailingToReconcile"
+            exp_labels:
+              severity: "critical"
+              operator_health_impact: "critical"
+              kubernetes_operator_part_of: "kubevirt"
+              kubernetes_operator_component: "ssp-operator"
+
+      # SSP pod is up, and reconciliation succeeded
+      - eval_time: "12m"
+        alertname: "SSPFailingToReconcile"
+        exp_alerts: [ ]
+
+  # SSPHighRateRejectedVms alert tests
+  - interval: "1m"
+    input_series:
+      - series: 'kubevirt_ssp_template_validator_rejected_total{pod="virt-template-validator-12345"}'
+        values: '0+1x10 10x120'
```

Comment:
`0+1x10 10` could work too, IIRC the last value is repeated indefinitely.

Reply:
Here the change did not work. I'm not sure why, probably there is some interpolation and `0+1x10 10` evaluates to a different value at 65 min time step.

Reply:
> 'a+bxn' becomes 'a a+b a+(2*b) a+(3*b) … a+(n*b)'
> Read this as series starts at a, then n further samples incrementing by b.

Weird. Also, if the last value is repeated, then `0+1x10` could work as well. It could be about the `increase` function in `TemplateValidatorRejectedIncreaseQuery`.

Reply:
Yes, probably.